### PR TITLE
[refs #42] Introduce our testing policy

### PR DESCRIPTION
**Pull request by AdamChrimes**
Add documentation for our testing policy for browsers, screen
resolutions and assistive technologies. Including a checklist for pull
requests. Also remove redundant documentation sections.

### CODE REVIEW

**Comment by AdamChrimes** [comment]
@mcheung-nhs @davidhunter08 is there anything you’d like to add/remove to this?

File: docs/contributing/testing.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+‘Compliant’ means that the components must look as good and function as well as they do in other modern browsers.
+
+'Functional' means the components may not look perfect in the given browser, but must still be usable without errors and without 'looking broken'.
+
+## Assistive technology support
+
+| Software                 | Type               | Browser                             |
+|--------------------------|--------------------|-------------------------------------|
+| JAWS                     | screen reader      | Internet Explorer 11                |                     
+| JAWS                     | screen reader      | Mozilla Firefox (latest 2 versions) |
```
Comment (mcheung-nhs) [inline docs/contributing/testing.md]: Don't need to test in JAWS on Firefox, IE is sufficient

**Comment by AdamChrimes** [comment]
@mcheung-nhs cheers Michael, is there anything you can add to the Accessibility section, maybe about testing for users that change colours of their screen/high contrast mode?

File: docs/contributing/testing.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+| Windows          | Mozilla Firefox (latest 2 versions)    | compliant   |
+| macOS            | Safari 9+                              | compliant   |
+| macOS            | Google Chrome (latest 2 versions)      | compliant   |
+| macOS            | Mozilla Firefox (latest 2 versions)    | compliant   |
+| iOS              | Safari for iOS 9.3 and later           | compliant   |
+| iOS              | Google Chrome (latest 2 versions)      | compliant   |
+| Android          | Google Chrome (latest 2 versions)      | compliant   |
+| Android          | Samsung Internet (latest 2 versions)   | compliant   |
+
+‘Compliant’ means that the components must look as good and function as well as they do in other modern browsers.
```
Comment (mcheung-nhs) [inline docs/contributing/testing.md]: Being pedantic... the difference in single quotes used in ‘Compliant’ and 'Functional' ;-)

**Comment by AdamChrimes** [comment]
@Ross-Clark @sabi98 have you any feedback on this? Does it all make sense?

**Comment by Ross-Clark** [comment]
@AdamChrimes I would consider adding links in the software column of the "Assistive Technology Support" section, linking to the respective softwares official website. This might make it easier for users of this page to research these peices of software if they are not familiar with them.

**Comment by AdamChrimes** [comment]
Good suggestion @Ross-Clark do you want to see if you can make the change? On the `feature/testing-criteria-docs` branch

**Pull request merged**